### PR TITLE
Interactive form

### PR DESCRIPTION
**Pull request by grace-mugoiri**
Connect CREATE_BOOK and DELETE_BOOK actions with the React components for dynamic web page.(milestone 3)

### CODE REVIEW

**Comment by grace-mugoiri** [comment]
Kindly review my work. I requested to work alone on this project.

File: src/components/Book.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/destructuring-assignment */
 /* eslint-disable react/prop-types */
 /* eslint-disable no-restricted-syntax */
 /* eslint-disable react/prefer-stateless-function */
```
Comment (bolah2009) [inline src/components/Book.js]: Let's resolve this disabled rules
- [`react/destructuring-assignment`] Let's destructure the props, we could do 
```
const { book: { id, title, category } } = this.props;
```
- [`react/prop-types`] Please check out https://reactjs.org/docs/typechecking-with-proptypes.html to typecheck your props

- [`react/prefer-stateless-function`] you can use a stateless function for this component inside of a class component

File: src/containers/BooksForm.js [hunk 2]
```diff
@@ -1,27 +1,78 @@
+/* eslint-disable react/destructuring-assignment */
 /* eslint-disable react/prefer-stateless-function */
 /* eslint-disable no-restricted-syntax */
```
Comment (bolah2009) [inline src/containers/BooksForm.js]: Please see the previous comment

File: src/containers/BooksForm.js [hunk 3]
```diff
@@ -1,27 +1,78 @@
...
+    const { name } = event.target;
+
+    this.setState({
+      [name]: event.target.value,
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    // eslint-disable-next-line react/prop-types
```
Comment (bolah2009) [inline src/containers/BooksForm.js]: See comment on type checking pros with `prop-types`

File: src/containers/BooksList.js [hunk 4]
```diff
@@ -1,16 +1,24 @@
+/* eslint-disable no-restricted-syntax */
 /* eslint-disable react/prop-types */
 /* eslint-disable react/destructuring-assignment */
```
Comment (bolah2009) [inline src/containers/BooksList.js]: See previous comments

**Review by bolah2009 — CHANGES_REQUESTED** [review]
Hi @grace-mugoiri 

Nice job, could you resolve the disabled linters? Many of them could be resolved with refactoring the code.


#### :recycle:  Make these changes and submit another code review request. 👍🏾

#### Feel free to reach me on slack if you having issues implementing these features. :relaxed: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Comment by bolah2009** [comment]
Please, also include the milestone this PR is in the PR description.

File: .eslintrc.json [hunk 5]
```diff
@@ -18,6 +18,31 @@
...
+	  "react/prop-types": 0,
+	  "import/extensions": 0,
+      "no-plusplus": ["error", { "allowForLoopAfterthoughts": true }],
+	  "max-classes-per-file": 0,
+	  "react/button-has-type": 0,
+	  "react/destructuring-assignment": 0,
+	  "react/no-access-state-in-setstate": 0,
+	  "react/no-array-index-key": 0,
+	  "react/no-unused-prop-types": 0,
+	  "react/jsx-props-no-spreading": 0,
```
Comment (bolah2009) [inline .eslintrc.json]: Please let's use the [**correct config files**](https://github.com/microverseinc/linters-config/tree/master/react-redux).

> Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))

File: src/components/Book.js (hunk 1 above)
Comment (bellom) [reply src/components/Book.js]: @grace-mugoiri, Good job making some corrections.
Kindly make the remaining few ones from the previous reviewer @bolah2009
I quote: 
```
[react/prefer-stateless-function] you can use a stateless function for this component inside of a class component.
```
```
[react/prop-types] Please check out https://reactjs.org/docs/typechecking-with-proptypes.html to type check your props.
```

File: src/components/Book.js [hunk 6]
```diff
@@ -1,16 +1,20 @@
-/* eslint-disable react/prop-types */
 /* eslint-disable no-restricted-syntax */
 /* eslint-disable react/prefer-stateless-function */
```
Comment (bellom) [inline src/components/Book.js:1]: As the previous reviewer mentioned, you can use stateless function for this.
That is;
```
const Book = props => {
  const { book:........ } = props;
  return (
  .......
}
```

File: .eslintrc.json [hunk 7]
```diff
@@ -17,7 +17,8 @@
 	  "react/jsx-filename-extension": ["warn", { "extensions": [".js", ".jsx"] }],
 	  "import/no-unresolved": "off",
 	  "no-shadow": "off",
-	  "arrow-parens":"off",
-	  "no-restricted-syntax": ["error", "FunctionExpression", "WithStatement", "BinaryExpression[operator='in']"]
+	  "arrow-parens": "off",
+	  "no-restricted-syntax": ["error", "FunctionExpression", "WithStatement", "BinaryExpression[operator='in']"],
+	  "react/prop-types": 0
```
Comment (bellom) [inline .eslintrc.json]: As the previous reviewer mentioned, you can check out https://reactjs.org/docs/typechecking-with-proptypes.html to typecheck your props
That is;
```
Book.propTypes = {
  book: PropTypes.......
}
```
After setting the prop-types, you can then remove the line 22 :+1:

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @grace-mugoiri  :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

**For some reason you didn't get to attend to some of the corrections pointed out by the previous reviewer. :disappointed:** 

**_Check my comments below_** 

**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by grace-mugoiri** [comment]
I have really tried to push using props without errors.

File: src/components/Book.js [hunk 8]
```diff
@@ -20,4 +21,14 @@ class Book extends React.Component {
...
 }
 
+Book.propTypes = {
+  book: PropTypes.shape({
+    id: PropTypes.number.isRequired,
+    title: PropTypes.string.isRequired,
+    category: PropTypes.string.isRequired,
+  }).isRequired,
+  handleBookRemoval: PropTypes.func.isRequired,
+  // removeBook: PropTypes.func.isRequired,
```
Comment (bellom) [inline src/components/Book.js]: I trust you can remove this comment before merging

**Review by bellom — APPROVED** [review]
## Hi @grace-mugoiri  :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**